### ISSUE TITLE
RunImageJMacro throws an ascii codec error in 4.2.6 Mac built app bundle

### ISSUE THREAD

**Issue by bethac07** [CellProfiler/CellProfiler#4858]
Hat tip to @emiglietta , who is making a successful campaign for my "finder of the weirdest CellProfiler bugs" crown.

**Describe the bug**
When trying to run a macro with RunImageJMacro in Mac, in 4.2.6 the .app, you get this error

```
Error while processing RunImageJMacro:
'ascii' codec can't decode byte 0xe2 in position 20: ordinal not in range(128)
```

This does not happen in 
4.2.4 built
4.2.5 built
4.2.x source
4.2.6 built, IF you open the app bundle and just directly run the `cp` executable

If you put the module in debug mode, the entire macro runs, because the output folder has the output in it. If the module isn't in debug mode, it does clean up, which I think should rule out an issue in the `finally` block (the button etc it makes works, and no difference if eye is open or shut). Which narrows it down really to [these 22 lines](https://github.com/CellProfiler/CellProfiler/blob/4.2.x/cellprofiler/modules/runimagejmacro.py#L295-L316). The `subprocess` call seems perhaps the most likely candidate for something that would run differently in the sandboxed app vs its executable? But I'm a bit flummoxed tbh.

**To Reproduce**
Run attached with basically any image in 4.2.6 built vs, like, anywhere else. Then, cry.
[RunImageJMacroWorkflow.zip](https://github.com/CellProfiler/CellProfiler/files/13982623/RunImageJMacroWorkflow.zip)

**Comment by bethac07** [comment]
I guess my other sneaking suspicion would be that something in the `os.path.exists` call gets weird, but, cmon, `os` and `subprocess` should really NOT be things that we need to worry about their behavior across python minor versions. They should just work... (FWIW, 3.8.15 in the built 4.2.5, 3.8.17 in 4.2.6, if someone wants to check)

**Comment by bethac07** [comment]
Possibly not literally related, but supports my 'subprocess being the culprit' theory. Probably wouldn't _hurt_ to defensively add the fix there for error encoding at least in master, and in 4.2.x if we ever think another release is likely.

I guess worth testing if error persists in the shiny new dev builds...

https://stackoverflow.com/questions/75378950/subprocess-run-gives-me-error-unicodedecodeerror-ascii-codec-cant-decode-by

**Comment by bethac07** [comment]
Does not happen in the new dev builds. So this is a pure 4.2.6 problem.

**Comment by bethac07** [comment]
Fixed in 4.2.7 built, at least on my machine
